src/System Application/App/Format Validation/src/FormatValidationImpl.Codeunit.al

```diff
@@ -0,0 +1,44 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+namespace System.Utilities;
+
+using System;
+
+codeunit 5054 "Format Validation Impl."
+{
+    Access = Internal;
+
+    var
+        PhoneNoCannotContainLettersErr: Label 'must not contain letters';
+
+    procedure ValidateIfPhoneNoContainsCharacters(PhoneNo: Text[30]): Boolean
+    var
+        Char: DotNet Char;
+        i: Integer;
+    begin
+        for i := 1 to StrLen(PhoneNo) do
+            if Char.IsLetter(PhoneNo[i]) then begin
+                exit(true)
+            end;
+
+        exit(false);
+    end;
+
+    procedure ThrowErrorIfPhoneNoContainsCharacters(PhoneNo: Text[30]; SourceRecordId: RecordId; SourceFieldId: Integer)
+    var
+        SourceRecordId: RecordId;
+        RecRef: RecordRef;
+        FieldRef: FieldRef;
+        SourceFieldId: Integer;
+    begin
+        if not ValidateIfPhoneNoContainsCharacters(PhoneNo: Text[30]) then
```

Comment:
This won't compile.
It should be:
```suggestion
        if not ValidateIfPhoneNoContainsCharacters(PhoneNo) then
```

Reply:
Solved. I did not test and did not compile yet because there are to many open links. I want to develop close to final before I want to take the next steps...